### ISSUE TITLE
[FR] make the date/time stamp of the 'compile' available to display. 

### ISSUE THREAD

**Issue by dbyy** [MarlinFirmware/Marlin#27589]
### Is your feature request related to a problem? Please describe.

no, not related to a problem.

### Are you looking for hardware support?

n/a

### Describe the feature you want

It would be nice to have the date/time stamp of the last compile (from the firmware.bin file) available to display on the printers screen when running the firmware.
When testing some features in Marlin,  I may end up with several compilations in a short period of time.
To have the date/time stamp available on the display helps to keep the different versions apart.
It would be a 'nice to have'.

### Additional context

I'm not thinking of a new '#define". More of an internal variable that can be displayed on the printers screen

**Comment by robherc** [comment]
Is there a reason why the compile datetimestamp from Serial Report isn't sufficient here?

**Comment by ellensp** [comment]
this is almost trivial

When BUILDDATE is defined add a submenu "Build Datetime" to the About Printer menu
This menu shows the build date and time
I didn't want to add it to the Printer info menu, as there is already a release date and time in there and it was just confusing.

Eg added submenu
![Screenshot from 2024-12-16 12-58-26](https://github.com/user-attachments/assets/9b1e973b-984d-49aa-87f0-b784ac8bb52f)

The build date and time
![Screenshot from 2024-12-16 12-57-11](https://github.com/user-attachments/assets/fe5b8104-92ea-4261-abd4-acc82b9df454)



```DIFF
diff --git a/Marlin/Configuration_adv.h b/Marlin/Configuration_adv.h
index 564f44df60..f992dd7884 100644
--- a/Marlin/Configuration_adv.h
+++ b/Marlin/Configuration_adv.h
@@ -1546,8 +1546,9 @@
   #endif
 
   // Include a page of printer information in the LCD Main Menu
-  //#define LCD_INFO_MENU
+  #define LCD_INFO_MENU
   #if ENABLED(LCD_INFO_MENU)
+    #define BUILDDATE
     //#define LCD_PRINTER_INFO_IS_BOOTSCREEN // Show bootscreen(s) instead of Printer Info pages
   #endif
 
diff --git a/Marlin/src/lcd/menu/menu_info.cpp b/Marlin/src/lcd/menu/menu_info.cpp
index 2db0caa805..81ca11beca 100644
--- a/Marlin/src/lcd/menu/menu_info.cpp
+++ b/Marlin/src/lcd/menu/menu_info.cpp
@@ -263,6 +263,18 @@ void menu_info_board() {
 
 #endif
 
+//
+// "Build Date" submenu
+//
+#if ENABLED(BUILDDATE)
+  void menu_info_build() {
+    if (ui.use_click()) return ui.go_back();
+    START_SCREEN();
+    STATIC_ITEM_F(F(__DATE__ " " __TIME__));
+    END_SCREEN();
+  }
+#endif
+
 //
 // "About Printer" submenu
 //
@@ -306,6 +318,11 @@ void menu_info() {
   }
   #endif
 
+  #if ENABLED(BUILDDATE)
+    SUBMENU(MSG_INFO_BUILD, menu_info_build);
+  #endif
+
+
   END_MENU();
 }
 
```

Everything needs a define, so it can be removed entirely from the code.

**Comment by dbyy** [comment]
This is awesome.
Just wondering, on which version of configuration/configuration_adv.h is the 'diff' based upon?
I'm using 'configs' of Marlin 2.1.2.4 with BTT-SKR-Mini-E3/BTT TF35 and can't locate the spots so easily in my files.
Thanks for the suggestions.

**Comment by ellensp** [comment]
based on bugfix 2.1.x

**Comment by ellensp** [comment]
this is for marlin menus, to update the touch interface would require you to update the btt tft firmware

**Comment by ellensp** [comment]
I have created a PR to add this

**Comment by dbyy** [comment]
> based on bugfix 2.1.x

I figured out where it has to go.
The only problem is (and I'm not a c++ guru) I'm receiving an error when compiling  `'Marlin/src/lcd/menu/menu_info.cpp: In function 'void menu_info_board()':
Marlin/src/lcd/menu/menu_info.cpp:232:5: error: a function-definition is not allowed here before '{' token  `
and later on as consequence of that
`Marlin/src/lcd/menu/menu_info.cpp:333:29: error: 'menu_info_build' was not declared in this scope; did you mean 'menu_info_board'?
  333 |     SUBMENU(MSG_INFO_BUILD, menu_info_build);               // Printer Stats >`
So at the moment I'm stuck with that. Too bad.
Any help is much appreciated even though its not really a problem of  the Marlin code.
TIA

**Comment by dbyy** [comment]
Update:
I figured it out.
I positioned the changes at the wrong spot in "menu_info.cpp".
After I consulted the original file from bug-fix 2.1.x, I changed the location (to the right spot), and now it's compiling w/o any error.
Have to flash the firmware and see how it looks on the display.
Thanks again for all feedback.
